.github/workflows/iroha2-pr-heavy.yml

```diff
@@ -1,72 +1,63 @@
-name: I2::Dev::Tests (Heavy)
+name: I2::Release::Tests
 
 on:
   pull_request:
     branches: [iroha2]
-    tags:
-      - "v*.*.*"
-    paths:
-      - "**.rs"
-      - "**.json"
-      - "**.toml"
-      - "**.yml"
-
-      # Not part of the workspace
-      - "!wasm/**"
 
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  test-docker:
-    runs-on: ubuntu-latest
+  cli-test:
+    runs-on: [self-hosted, linux]
     container:
       image: 7272721/i2-ci:latest
     steps:
       - uses: actions/checkout@v2
       - uses: Swatinem/rust-cache@v1
-      - name: Set up Docker Buildx
-        id: buildx
-        uses: docker/setup-buildx-action@v1
+      # - name: Set up Docker Buildx
+      #   id: buildx
+      #   uses: docker/setup-buildx-action@v1
       - name: Build Client CLI
         run: mold -run cargo build
         working-directory: client_cli
-      - name: Build and push Iroha Docker image
-        uses: docker/build-push-action@v2
-        with:
-          context: .
-          push: false
-          tags: hyperledger/iroha2:dev
-
-      - name: Setup docker test environment
-        run: ./scripts/setup_docker_test_env.sh
-      - name: Docker compose genesis test
-        run: bash -c './scripts/test_genesis_docker_compose.sh || ( docker-compose logs; false )'
-      - name: Docker compose test
-        run: bash -c './scripts/test_docker_compose.sh || ( docker-compose logs; false )'
-      - name: Cleanup docker test environment
-        run: ./scripts/cleanup_docker_test_env.sh
+      # - name: Build and push Iroha Docker image
+      #   uses: docker/build-push-action@v2
+      #   with:
+      #     context: .
+      #     push: false
+      #     tags: hyperledger/iroha2:dev
+      - name: Build Iroha Peer
+        run: mold -run cargo build
+        working-directory: cli
+      - name: Mark both binaries as executable
+        run: chmod +x target/debug/iroha && chmod +x target/debug/iroha_client_cli
+      - name: Setup test environment
+        run: bash -c './scripts/test_env.sh setup'
+      - name: Genesis test
+        run: bash -c './scripts/tests/genesis.sh || (cat test/peers/iroha0.log; false )'
+      - name: Basic register and mint
+        run: bash -c './scripts/tests/register_mint_quantity.sh || (cat test/peers/iroha0.log; false )'
```

Comment:
it would be better if this was done in some sort of a loop (for every bash script in `./scripts/tests/`) instead of listing them manually

Reply:
why are the two test scripts empty?

Reply:
> it would be better if this was done in some sort of a loop (for every bash script in ./scripts/tests/) instead of listing them manually

I didn't do it because I wanted to have these test cases as separate steps in a job, so we have a nice idiomatic error report given by the CI, and not emulated in the logs by bash. 

> why are the two test scripts empty?

They're not supposed to be. I think this is because I needed to change their mode, and GitHub doesn't show the contents in that case... I'll double check regardless.

Reply:
I think, that running new test when a new file was added in `scripts/tests/` would be very implicit. Especially for review of such PR

Reply:
>>   why are the two test scripts empty?

Just double checked. The files are not empty. 

>> I think, that running new test when a new file was added in scripts/tests/ would be very implicit. Especially for review of such PR

You could create a separate workflow that added a tag if someone touched anything in that folder. 

With that said, I think that the `for` loop would be a premature optimization.